Knack Conversion for consumption module

williexu opened the pull request:
Closes https://github.com/Azure/azure-cli/issues/4928

-changed tests to record-only

azuresdkci commented:
View a preview at https://prompt.ws/r/Azure/azure-cli/5082
This is an **experimental preview** for @microsoft.com users.
(It may take a minute or two for your instance to be ready)
Email feedback to 'azfeedback' with subject 'Prompt Feedback'.

tjprescott commented on src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_params.py:
```diff
@@ -4,12 +4,14 @@
+        c.argument('include_meter_details', options_list=('--include-meter-details', '-m'), action='store_true', help='include meter details in the usages')
+        c.argument('start_date', options_list=('--start-date', '-s'), type=get_datetime_type(), help='start date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
+        c.argument('end_date', options_list=('--end-date', '-e'), type=get_datetime_type(), help='end date (in UTC Y-m-d) of the usages. Both start date and end date need to be supplied or neither')
+        c.argument('billing_period_name', options_list=('--billing-period-name', '-p'), help='name of a specific billing period to get the usage details that associate with')
```
Please convert the tuple syntax for `options_list` to list syntax.

williexu commented:
Just manually checked the commands on my own subscription, and they work perfectly. Tests were changed to record_only() since `start_date`/`end_date`/`billing_period` arguments vary between subscriptions (and some subscriptions dont support consumption). Tests can be run live provided a supported subscription is used and the tests are changed to reflect the right dates/periods.

@tjprescott  addressed your comment

tjprescott commented:
Looks good. Just get the CI green :)
